### PR TITLE
[freeboxos] Add FTTH and xDSL line status

### LINKED ISSUE #17073: [freeboxos] Channel xdsl-status not found on revolution thing
The channel xdsl-status is documented in README but not found in revolution thing.
This channel was present in the original binding and the main channel I was using.

### PR DESCRIPTION
**Pull request by clinique**
Resolves #17073 

I've tested the FTTH side and starting the coding of xDSL part.

### CODE REVIEW

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java:117]: Except if the core framework properly handles no change of channels, this is better to check if at least one channel was added or removed.
Core framework to be checked.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:504]: Rather "dBm" instead of "%unit%".

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:503]: Not sure. If I correctly remember, QualityOfService expects few discrete values to render the proper icon.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:55]: Can you please add upgrade instructions ?

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:468]: To be checked but I believe we never set such category while the item type is Switch.
This will be the default value anyway.

Same for other added channel types.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:495]: Link Is Active

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:75]: Connection xDSL Status

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:77]: I will rather suggest a new channel type linked to a String and we will map `SynchroState`.
You can simply remove this channel and I will add it myself.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java:117]: I did this previously because only dynamic (sensors, fans...) channels were added.
Now for sure channels will be removed depending upon the type of media used (FTTH, xDSL).
So you can be sure that at this point channels will have changed.

**Comment by lolodomo** [comment]
One improvement to do: we should avoid calling an API if we know that this call is only to set channels and none of these channels are linked. You could start by doing it for `getFtthStatus`.

PS: it should be checked and if necessary fixed in all the binding. Of course not in that PR.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:77]: Can you directly include your code modifications in your PR to this branch ? I will not be able to proceed by myself in the next coming days.

Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:77]: You mean submit a PR to your branch ? Yes I can but in that case we will have to wait for another maintainer to merge "our" PR.
My initial idea was to merge your PR and then create another one to handle xDSL. Here is why I suggested to remove this channel.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:504]: Done

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:503]: The icon is not dynamic but properly rendered.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:468]: Would you prefer contact instead of switch ? They are read-only in any case.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:75]: Changed

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:77]: I'm going to proceed further with this PR and you'll add a second one.
The 'status' channel is temporary until you add yours. If there is not channel in the group it shows an error.

**Comment by clinique** [comment]
> @clinique : please rebase and solves the conflicts.

I'm on it

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java:195]: Move this block outside the parent block.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/server-channel-groups.xml:77]: Ok

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:467]: No, keep switch please.
I was wrong, this is sometimes provided like that, for example in the old Freebox binding.
So, let's keep this category.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:502]: Not fully convinced but ok let's keep this category.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java:195]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**
